vespa/application.py

```diff
@@ -802,6 +796,35 @@ def get_document_v1_path(
         return f"/document/v1/{namespace}/{schema}/docid/{id}"
 
 
+class CustomHTTPAdapter(HTTPAdapter):
+    def __init__(self, *args, **kwargs):
+        self.pool_maxsize = kwargs.pop("pool_maxsize")
+        self.pool_connections = kwargs.pop("pool_connections")
+        self.num_retries_429 = kwargs.pop("num_retries_429", 10)
+        super().__init__(*args, **kwargs)
+
+        retry_strategy = Retry(
+            total=3,
+            backoff_factor=1,
+            raise_on_status=True,  # raise exception on status
+            status_forcelist=[503],
+            allowed_methods=["POST", "GET", "DELETE", "PUT"],
+        )
+        self.retry_strategy = retry_strategy
+
+    def send(self, request, **kwargs) -> Response:
+        for attempt in range(self.num_retries_429):
+            response = super().send(request, **kwargs)
+            if response.status_code == 429:
+                wait_time = (
+                    2** attempt + random.uniform(0, 1)
+                )  # Exponential backoff with jitter. The 10th retry will sleep for 1024 seconds.
```

Comment:
I think that is a bit excessive, perhaps cap it at 10s or so, and even start a bit lower?

Reply:
Did we decide to reduce this a bit, though? I can't remember now 😁